File: monitoring/src/main/java/com/example/monitoring/user/application/port/in/request/CreateUserRequestDto.java
package com.example.monitoring.user.application.port.in.request;

import com.example.monitoring.user.domain.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreateUserRequestDto {

    private String name;
    private int age;

    public User createUser() {
        return User.builder()
                .name(name)
                .age(age)
                .build();
    }
}

File: kotlin-spring-boot/src/main/kotlin/com/hello/kotlin/user/application/request/CreateUserRequestDto.kt
package com.hello.kotlin.user.application.request

import com.hello.kotlin.user.domain.User

data class CreateUserRequestDto (
    val name: String,
    val age: Int,
    val address: String,
) {
    fun toEntity() = User(
        name,
        age,
        address
    )
}